product_docs/docs/epas/17/reference/oracle_compatibility_reference/epas_compat_sql/65b_import_foreign_schema.mdx

```diff
@@ -0,0 +1,86 @@
+---
+title: "IMPORT FOREIGN SCHEMA"
+---
+
+## Name
+
+`IMPORT FOREIGN SCHEMA` &mdash; Import a foreign schema from a foreign server or database link.
+
+## Synopsis and parameters
+
+See [IMPORT FOREIGN SCHEMA](https://www.postgresql.org/docs/current/sql-importforeignschema.html) for command syntax, parameters and functionality. 
+
+NOT NULL constraints are imported by  default. Use the `import_not_null` option if you don’t want to import NOT NULL constraints.
+
+## Description
+
+After you [create a public database link](../../../reference/oracle_compatibility_reference/epas_compat_sql/21_create_public_database_link/), you can import a foreign schema with `IMPORT FOREIGN SCHEMA`. You can import foreign schemas with database links created with `edb_dblink_oci` connections as explained in the [database link creation examples](../../../reference/oracle_compatibility_reference/epas_compat_sql/21_create_public_database_link/#examples).
+
+You can then access the imported schema objects locally, instead of remotely.
+
+## Example 
+
+This example uses the `IMPORT FOREIGN SCHEMA` command to import a database schema named `IMP` from an Oracle server to an EDB Postgres Advanced Server. The target schema with the imported table definitions is called `tmp_schema` and uses an `edb_dblink_oci` connection called `oci_link`.
+
+### Prerequisites
+
+- [Create a public database link](../../../reference/oracle_compatibility_reference/epas_compat_sql/21_create_public_database_link/)
+
+- Ensure you have the `connstr` and `foreign data wrapper` names at hand.
+
+### Import schemas 
+
+1.  [Create a server](https://www.postgresql.org/docs/current/sql-createserver.html) (here called `oci_link`) that connects you to the Oracle database. Specify the foreign data wrapper, and the connection string required to reach the Oracle service.
+
+    ```sql
+    CREATE SERVER oci_link FOREIGN DATA WRAPPER link_fdw OPTIONS (connstr :ORACLE_SERVICE);
+    ```
+
+1.  Map the EDB Postgres `edb` user to the Oracle database user for which you want to import the schemas. Use the `oci_link` connection created previously, and specify the Oracle’s user and password:
+
+    ```sql
+    CREATE USER MAPPING FOR edb SERVER oci_link OPTIONS (password c, "user" :ORACLE_USER_NAME);
```

Comment:
`:ORACLE_USER_NAME);` looks like an env variable. Should we replace it with a placeholder so it is easier for the reader to understand this is a value to replace? E.g. with `<oracle_username>`.

Reply:
Same as above